CMakeLists.txt

```diff
@@ -10,7 +10,7 @@
 ###############################################################################
 # General settings
 ###############################################################################
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.9 FATAL_ERROR)
```

Comment:
This is a big jump. Maybe it's time to finally leave CMake 2.8 behind, but I wonder if it would be possible to optionally have the LTO option only when CMake version is high enough while leaving this requirement unchanged for now? This would make the decision about merging this PR very easy to make.

Reply:
I'm not sure.
I will investigate further.

Reply:
As far as I know, [CheckIPOSupported]  and [CMAKE_INTERPROCEDURAL_OPTIMIZATION] which are the official, 
portable support for checking and enabling LTO are only available in 3.9 and later.

If we want to do it without requiring 3.9, we would have to manually write this by ourselves.

[CheckIPOSupported]: https://cmake.org/cmake/help/latest/module/CheckIPOSupported.html
[CMAKE_INTERPROCEDURAL_OPTIMIZATION]: https://cmake.org/cmake/help/latest/variable/CMAKE_INTERPROCEDURAL_OPTIMIZATION.html

Reply:
However, I did find an existing cmake script for checking lto support [here] that might be helpful.

[here]: https://github.com/wang-bin/cmake-tools/blob/46865006323046a3be0646373652087a91634aea/tools.cmake#L435

Reply:
@vadz I've found a solution:

I put a separate `cmake_minimum_required` check for the LTO code which are executed only when `SOCI_LTO` is set to 
`ON`.

The global cmake version is restored to 2.8

Reply:
Only when someone wants LTO support, will they have to use cmake 3.9